test_scripts/WebEngine/commonWebEngine.lua

```diff
@@ -0,0 +1,285 @@
+---------------------------------------------------------------------------------------------------
+-- Common module
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local actions = require("user_modules/sequences/actions")
+local utils = require("user_modules/utils")
+local runner = require('user_modules/script_runner')
+local events = require('events')
+local test = require("user_modules/dummy_connecttest")
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+config.defaultProtocolVersion = 2
+
+--[[ Shared Functions ]]
+local common = {}
+common.Title = runner.Title
+common.Step = runner.Step
+common.start = actions.start
+common.stopSDL = actions.sdl.stop
+common.getHMIConnection = actions.hmi.getConnection
+common.registerApp = actions.registerApp
+common.registerAppWOPTU = actions.registerAppWOPTU
+common.activateApp = actions.activateApp
+common.getMobileSession = actions.getMobileSession
+common.cloneTable = utils.cloneTable
+common.printTable = utils.printTable
+common.tableToString = utils.tableToString
+common.isTableEqual = utils.isTableEqual
+common.wait = actions.run.wait
+common.getAppsCount = actions.getAppsCount
+common.cprint = utils.cprint
+common.getConfigAppParams = actions.getConfigAppParams
+common.policyTableUpdate = actions.policyTableUpdate
+common.ptsTable = actions.sdl.getPTS
+common.isPTUStarted = actions.isPTUStarted
+common.failTestStep = actions.run.fail
+common.getPreloadedPT = actions.sdl.getPreloadedPT
+common.setPreloadedPT = actions.sdl.setPreloadedPT
+common.null = actions.json.null
+common.EMPTY_ARRAY = actions.json.EMPTY_ARRAY
+common.sdlBuildOptions = test.sdlBuildOptions
+common.deletePTS = actions.sdl.deletePTS
+common.runAfter = actions.run.runAfter
+common.unRegisterApp = actions.app.unRegister
+common.backupSDLIniFile = actions.sdl.backupSDLIniFile
+common.setSDLIniParameter = actions.sdl.setSDLIniParameter
+
+--[[ Local Variables ]]
+common.defaultAppProperties = {
+  nicknames = { "Test Web Application_1", "Test Web Application_2" },
+  policyAppID = "0000001",
+  enabled = true,
+  authToken = "ABCD12345",
+  transportType = "WS",
+  hybridAppPreference = "CLOUD",
+  endpoint = "ws://127.0.0.1:8080/"
```

Comment:
I dont think there is supposed to be an endpoint for webengine applications. This looks like a cloud app policy table entry.

Reply:
@JackLivio Yes, that's right - `endpoint` field is for cloud applications. However we added it in order to check possibility to set all available attributes through `SetAppProperties` RPC. Having this field defined has no impact on tests and Application can be connected through any of available transports - TCP, WS or WSS.